### DISCUSSION TITLE
grapher kwargs modification

### DISCUSSION THREAD

**Discussion by fdion** [adamerose/pandasgui#90] [Q&A]
When I implemented symbols for lines, I relied on what was already there, namely "data_frame" in kwargs. This was defined in `on_dragger_finished` as `kwargs = {"data_frame": df}`in grapher.py.

In the current develop branch, I see kwargs is now `kwargs = {}` in `on_dragger_finished`. Nothing else sets it.

I'm not sure if that was needed for some other part of the code, but this break:
- line plots
- bar plots

Maybe that was an artifact of a merge? Trying to figure out wheter that is the bug, or if we need to fix line and bar in a different way, or is there something else missing somewhere else.

Figured I'd start a discussion instead of a bug, cause I don't know exactly how to formulate the bug report. Also not sure what section I should put this in, so I'm putting it in Q&A

**Comment by adamerose** [comment]
Looks like I broke this in one of my commits adding code history support to the Grapher. I needed to be able to call `.add_history_item` inside the graphing functions to record the equivalent Python snippet for each one, this meant passing a `PandasGuiDataFrameStore` object to the plotting functions instead of a simple DataFrame.

To work with this new structure, I access the DataFrame inside the plot functions using `pgdf.df` instead of `kwargs['data_frame']`, just forgot to update the logic you added to line and bar. I think I just fixed it now though if you want to double check.

**Comment by fdion** [comment]
I figured something needed more flexibility and/or multiple calls.

In regards to line, this works fine, but for the bar plot, `key_cols` is not what you'd think and it is throwing a `KeyError`. Using one of your demo dataset, selecting `Speed` for `x` you get not `Speed` in `key_cols`, but `('S', 'p', 'e', 'e', 'd')`. Are there hierarchical kwargs passed to `bar`?

**Reply by adamerose** [reply]
Yeah if you have multiple columns dropped in a category like `x` then the value in `kwargs` will be a list of strings instead of a string. The logic at the top of `line` accounts for this and I tried simplifying it to a one-liner in `bar` but apparently didn't work.
